sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/ecert-integration/_tests_/ecert-full-time-process-integration.scheduler.e2e-spec.ts

```diff
@@ -834,6 +837,111 @@ describe(
       expect(isScheduleNotSent).toBe(true);
     });
 
+    it("Should not generate disbursement when the Student answered application question 'estranged from the parents' as 'yes' and profile does not have an approved modified independent status.", async () => {
+      // Arrange
+      const modifiedIndependentStatus = ModifiedIndependentStatus.NotRequested;
+      const estrangedFromParents = FormYesNoOptions.Yes;
+      const application = await saveApplicationDisbursementForECertGeneration({
+        modifiedIndependentStatus,
+        estrangedFromParents,
+      });
+
+      // Queued job.
+      const mockedJob = mockBullJob<void>();
+
+      // Act
+      const result = await processor.processQueue(mockedJob.job);
+
+      // Assert uploaded file.
+      const uploadedFile = getUploadedFile(sftpClientMock);
+      const uploadedFileName = getUploadedFileName();
+      expect(uploadedFile.remoteFilePath).toBe(uploadedFileName);
+      expect(result).toStrictEqual([
+        `Generated file: ${uploadedFileName}`,
+        "Uploaded records: 0",
+      ]);
+      expect(
+        mockedJob.containLogMessage(
+          `Student answered '${estrangedFromParents}' for estranged from parents but his modified independent status is '${modifiedIndependentStatus}', the disbursement calculation will not proceed.`,
+        ),
+      ).toBe(true);
+      const [disbursement] =
+        application.currentAssessment.disbursementSchedules;
+      const isScheduleNotSent = await db.disbursementSchedule.exists({
+        where: {
+          id: disbursement.id,
+          disbursementScheduleStatus: DisbursementScheduleStatus.Pending,
+        },
+      });
+      expect(isScheduleNotSent).toBe(true);
+    });
+
+    it("Should generate disbursement when the Student answered application question 'estranged from the parents' as 'yes' and profile has an approved modified independent status.", async () => {
+      // Arrange
+      const modifiedIndependentStatus = ModifiedIndependentStatus.Approved;
+      const application = await saveApplicationDisbursementForECertGeneration({
+        modifiedIndependentStatus,
+        estrangedFromParents: FormYesNoOptions.Yes,
+      });
```

Comment:
Creation of this factory, made it easy to focus on core part of the test. We should do more like this.